src/primitives/common/plane_sector.rs

```diff
@@ -14,65 +24,59 @@ use crate::{
 /// half-planes.
 #[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
 pub struct PlaneSector {
-    line_a: LinearEquation<Real>,
-    line_b: LinearEquation<Real>,
-    side_a: LineSide,
-    side_b: LineSide,
-    sweep: Angle,
+    /// Half plane on the left side of a line.
+    half_plane_left: LinearEquation,
+    /// Half plane on the right side of a line.
+    half_plane_right: LinearEquation,
+    /// The operation used to combine the two half planes.
+    operation: Operation,
 }
 
 impl PlaneSector {
-    pub fn new(center_2x: Point, angle_start: Angle, angle_sweep: Angle) -> Self {
-        let angle_end = angle_start + angle_sweep;
-
-        let angle_start_norm = angle_start.normalize_from(-ANGLE_90DEG);
-        let angle_end_norm = angle_end.normalize_from(-ANGLE_90DEG);
-        let negative_sweep = angle_sweep < Angle::zero();
+    pub fn new(center_2x: Point, mut angle_start: Angle, angle_sweep: Angle) -> Self {
+        let mut angle_end = angle_start + angle_sweep;
 
-        let side_a = if (angle_start_norm < ANGLE_90DEG) ^ negative_sweep {
-            LineSide::Left
-        } else {
-            LineSide::Right
-        };
+        // Swap angles for negative sweeps to use the correct sides of the half planes.
+        if angle_sweep < Angle::zero() {
+            core::mem::swap(&mut angle_start, &mut angle_end)
+        }
 
-        let side_b = if (angle_end_norm >= ANGLE_90DEG) ^ negative_sweep {
-            LineSide::Left
+        let angle_sweep_abs = angle_sweep.abs();
+        let operation = if angle_sweep_abs < ANGLE_180DEG {
+            Operation::Intersection
+        } else if angle_sweep_abs < ANGLE_360DEG {
+            Operation::Union
         } else {
-            LineSide::Right
+            Operation::EntirePlane
         };
 
         Self {
-            line_a: LinearEquation::from_point_angle(center_2x, angle_start),
-            line_b: LinearEquation::from_point_angle(center_2x, angle_end),
-            side_a,
-            side_b,
-            sweep: angle_sweep.abs(),
+            half_plane_left: LinearEquation::from_point_angle(center_2x, angle_start),
+            half_plane_right: LinearEquation::from_point_angle(center_2x, angle_end),
+            operation,
         }
     }
 
+    /// TODO: This method doesn't really return an empty plane sector. Does this matter?
```

Comment:
> Does this matter?

Not really. From looking at its usage in `PlaneSectorIterator::empty`, the intent is clear from that context so I'd be happy to keep this as `empty`. If you do want to change the name, would `zero` make sense? `infinity`?

Reply:
I've removed the method and replaced its only use with a call to `PlaneSector::new`. `PlaneSectorIterator::empty` now uses a `PlaneSector` with `operation == Operation::EntirePlane`, which is the fastest to construct and doesn't matter because the points iter is also empty.